csrc/optimization/alias_analysis.cpp

```diff
@@ -295,16 +295,17 @@ void AliasFinder::handle(const SliceOp* slice) {
   for (auto i = static_cast<int64_t>(out_rank) - 1; i >= 0; i--) {
     if (out_layout.allocation_domain[i]->isBroadcast()) {
       out_layout.contiguity[i] = std::nullopt;
-      continue;
-    }
-
-    if (next_non_broadcast_is_non_contiguous) {
+      // But don't reset next_non_broadcast_is_non_contiguous. Say we slice a
+      // TV of shape [4,1,7],[t,t,t] to a TV of shape [4,1,5]. The output
+      // contiguity should be [f,n,t].
+    } else if (next_non_broadcast_is_non_contiguous) {
```

Comment:
The change here does make sense to me. But I'm not following the comment here.

> TV of shape [4,1,7],[t,t,t]
> don't reset next_non_broadcast_is_non_contiguous.

The old code calls `continue` for broadcast dimension, so we are not skipping anything here are we?

But rather the broadcast dimension here is not viewed as such, since its contiguity flag is marked as `True`?

Reply:
Note this is over the `[f,n,t]` of the *output* tensor, Jie, not the `[t,t,t]` of the input tensor.
(but still a valid point, seems like the code is the same)

Reply:
I guess I see how the first example below when we are slicing a broadcast dimension, how this code could fix it.

But the second example where the comment is referring doesn't make sense to me at all.

Reply:
I added this comment because I almost wrote something like
```
    if (out_layout.allocation_domain[i]->isBroadcast()) {
      out_layout.contiguity[i] = std::nullopt;
      next_non_broadcast_is_non_contiguous = false;
    } else ...
```
which is wrong. We shouldn't reset `next_non_broadcast_is_non_contiguous` here when deciding the contiguity of a broadcast dimension. 

After a second thought, I feel it's pretty obvious because the variable name says it all, so I may just remove this comment.

Reply:
> But the second example where the comment is referring doesn't make sense to me at all.

How so? You think the contiguity ought to be something else?

Reply:
Ha, now I see where the comment it coming from.

I somehow was reading too much into the example and how those were translating to the code. NVM my poor reading.